src/main/java/com/atlassian/db/replica/internal/ReplicaStatement.java

```diff
@@ -558,15 +558,17 @@ public Statement getReadStatement(RouteDecisionBuilder decisionBuilder) {
             connectionProvider.getStateDecision().ifPresent(decisionBuilder::cause);
             return prepareWriteStatement(decisionBuilder);
         }
-        SqlQuery sqlQuery = new SqlQuery(decisionBuilder.getSql());
-        if (sqlQuery.isWriteOperation(sqlFunction)) {
-            decisionBuilder.reason(WRITE_OPERATION);
-            return prepareWriteStatement(decisionBuilder);
-        }
-
-        if (sqlQuery.isSelectForUpdate()) {
-            decisionBuilder.reason(LOCK);
-            return prepareWriteStatement(decisionBuilder);
+        String sql = decisionBuilder.getSql();
+        if (sql != null) {
```

Comment:
`ReplicaStatementIT` covers triggers the new `if` by calling `statement.executeQuery("SELECT 1;");` in #79 . I didn't add that test here to avoid conflicts.

Reply:
The only way to make this `null` is to call `getReadStatement` directly. But it's only called internally. It could be `protected` (maybe even `private`). I don't see a reason for it being public. PS. `getWriteStatement` + `getCurrentStatement` could also become `protected` at most.

Reply:
>ReplicaStatementIT covers triggers the new if by calling statement.executeQuery("SELECT 1;");

Yes. But for this case, SQL != null so that it will be green before and after.